### ISSUE TITLE
4.20.4 Issue with monotonic order and ZigZag Pattern

### ISSUE THREAD

**Issue by 5axes** [smartavionics/Cura#184]
I have an issue with your release of Cura , it's not a particular problem related to the last release, but I never notice this problem before (It's also because I never use the ZigZag Pattern).

Conditions : 
Monotonic Top/Bottom Order : On
Top/Bottom Pattern : ZigZag.

In this case  the pattern generate a lot of movement from one side to the other of the layer.
![image](https://user-images.githubusercontent.com/11015345/188010061-fb032b86-6a3e-44f1-a589-180a3566ad30.png)

On the standard release of Cura the Monotonic order is correct.
![image](https://user-images.githubusercontent.com/11015345/188010412-ec81e4c2-f9da-4a7a-b461-9ab690284bb9.png)

Sample Case 
[OneContour.zip](https://github.com/smartavionics/Cura/files/9473316/OneContour.zip)

**Comment by smartavionics** [comment]
When you say "standard release" which version are you referring to? Remember, my builds are based on 4.13 and not 5.

**Comment by 5axes** [comment]
Yes I'm Sorry. You're Right the 4.13 & 4.12 release of Ultimaker have the same Issue.

Solved in 5.1  ( Still a mess in 5.0)

**Comment by smartavionics** [comment]
No problem, I'll see if I can work out what they did to fix it and, if possible, will incorporate the fix.

**Comment by smartavionics** [comment]
Looks like this problem hasn't really been resolved yet https://github.com/Ultimaker/Cura/issues/13456

**Comment by 5axes** [comment]
Not sure , in this issue we have a problem with the ZigZag pattern. In the mentionned issue we have a linear pattern. Could be a different problem.

**Comment by smartavionics** [comment]
Oh, yeah, you're right. Let's see how they fix the latest. I never did manage to find what they had changed to fix the earlier issue.
